classes/Route.php

```diff
@@ -28,5 +28,12 @@ public function __construct(
 
         public ?Closure $user_check = null,
         public bool $dev_only = false,
+
+        // Layout controls
+        public bool $render_header = true,
+        public bool $render_sidebar = true,
+        public bool $render_topbar = true,
+        public bool $render_content = true,
+        public bool $render_footer = true
```

Comment:
This does not seem like something that's really route-specific. Homepage is a special case but unless we have multiple cases, let's not add these as extra route parameters, it's okay for the homepage to get hardcoded special treatment